Support more recent versions of megaparsec

https://github.com/haskell-hvr/HsYAML/blob/777165d2228e2c23145bb0e1ff5b968329b4ad34/HsYAML.cabal#L115

Right now it's a bit confusing when you try to build this package from source.

This requires adapting the code accordingly, due to

```
$ cabal build --allow-newer=HsYAML:megaparsec

Build profile: -w ghc-8.6.5 -O1
In order, the following will be built (use -v for more details):
 - HsYAML-0.2.0.0 (exe:yaml-test) (first run)
Preprocessing executable 'yaml-test' for HsYAML-0.2.0.0..
Building executable 'yaml-test' for HsYAML-0.2.0.0..
[1 of 2] Compiling TML              ( src-test/TML.hs, /stuff3/work/GitHub/My/HsYAML/dist-newstyle/build/x86_64-linux/ghc-8.6.5/HsYAML-0.2.0.0/x/yaml-test/build/yaml-test/yaml-test-tmp/TML.o ) [Text.Megaparsec.Expr changed]

src-test/TML.hs:43:1: error:
    Could not load module ‘Text.Megaparsec.Expr’
    It is a member of the hidden package ‘megaparsec-6.5.0’.
    Perhaps you need to add ‘megaparsec’ to the build-depends in your .cabal file.
    It is a member of the hidden package ‘megaparsec-6.5.0’.
    Perhaps you need to add ‘megaparsec’ to the build-depends in your .cabal file.
    It is a member of the hidden package ‘megaparsec-6.5.0’.
    Perhaps you need to add ‘megaparsec’ to the build-depends in your .cabal file.
    Use -v to see a list of the files searched for.
   |
43 | import           Text.Megaparsec.Expr
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

A PR would be highly appreciated :-)

--- a/HsYAML.cabal
+++ b/HsYAML.cabal
@@ -113,7 +113,7 @@ executable yaml-test
                  , containers
                  , mtl
                    -- non-inherited
-                 , megaparsec  >= 6.5.0 && < 6.6
+                 , megaparsec  == 7.0.*
                  , microaeson  == 0.1.*
                  , filepath    == 1.4.*
                  , directory   >= 1.2 && < 1.4

--- a/src-test/TML.hs
+++ b/src-test/TML.hs
@@ -40,12 +40,11 @@ import           System.Environment
 import           Text.Megaparsec
 import           Text.Megaparsec.Char
 import qualified Text.Megaparsec.Char.Lexer as L
-import           Text.Megaparsec.Expr
 
 type Parser = Parsec Void T.Text
 
 parse :: String -> T.Text -> Either T.Text Document
-parse fn raw = either (Left . T.pack . parseErrorPretty' raw)
+parse fn raw = either (Left . T.pack . errorBundlePretty)
                       (Right .process_pseudo)
                       (Text.Megaparsec.parse testml_document fn raw)
 
@@ -318,7 +317,7 @@ data_section = many block_definition
     block_definition = do
       -- block_heading
       string "===" *> ws
-      l <- T.pack <$> manyTill anyChar eol
+      l <- T.pack <$> manyTill anySingle eol
 
       -- TODO: user_defined
       ps <- many point_definition
@@ -333,7 +332,7 @@ data_section = many block_definition
 
       let single = do
             _ <- char ':' *> ws
-            x <- T.pack <$> manyTill anyChar eol
+            x <- T.pack <$> manyTill anySingle eol
             -- consume and ignore any point_lines
             _ <- point_lines
             pure $! case j of
@@ -358,7 +357,7 @@ data_section = many block_definition
 point_lines :: Parser T.Text
 point_lines = T.pack . unlines <$> go
   where
-    go = many (notFollowedBy point_boundary *> manyTill anyChar eol)
+    go = many (notFollowedBy point_boundary *> manyTill anySingle eol)
 
     point_boundary :: Parser ()
     point_boundary = void (string "---") <|> void (string "===") <|> eof

